src/types/line.js

```diff
@@ -243,6 +243,29 @@ module.exports = function(Chart) {
 			ctx.lineTo(view.x2, view.y2);
 			ctx.stroke();
 
+			ctx.restore();
+		},
+		drawLabel: function() {
```

Comment:
Is the label also being drawn in `draw` and then we're drawing it again here on top a second time? It looks like there's a lot of code duplication here

Reply:
I believe I have all of the label code pulled into `drawLabel.` However, there is some setup duplication as I'm not 100% confident in what is needed for drawing given that we're doing this in two passes (that will be broken up by a separate drawing operation, potentially resetting our values).

Reply:
At the very least I'd rather have `draw` call `drawLabel` rather than duplicate the code. But even that seems a little redundant to me

What values are you worried about being reset? I've never actually used this plugin, but just help out a bit in getting the code reviewed

Reply:
That would re-introduce the original issue here. The problem is that lines and labels are currently drawn at the same time, in order - so any following annotation is drawn on top of the previous ones. We need a separate loop so we draw all lines first, and all labels second, so lines are not atop labels. Since we're iterating twice, all of our values (position, size etc.) will be stomped in `ctx` by further iterations of the loop.

Reply:
There's no reason to copy/paste code though. You could call `drawLabel` from `draw` and still call `drawLabel` a second time from another for loop afterwards as you're doing now.

Reply:
Then you're just duplicating the draw action to no benefit.

Reply:
I don't understand. Aren't you already duplicating it by copy/pasting all the code? The benefit would be drawing the labels on top of the lines...

Reply:
The duplication is because you need the same context at a different iteration of the loop where the context is lost.  You could create a new setup function and use it both times to mutate the context but that seems like a lot of extra overhead and passing of state for little benefit.  DRY seems less important than readability in this case to me.

Reply:
Resolving this comment. I got a big confused by the way GitHub was displaying the diff originally.